server/app/app.py

```diff
@@ -353,32 +363,40 @@ def __init__(self, app_config):
         self.app.register_blueprint(webbp)
 
         api_version = "/api/v0.2"
+        backend_base_url = server_config.get_backend_base_url()
+        backend_path = "/"
+        if backend_base_url:
+            parse = urlparse(backend_base_url)
+            backend_path = parse.path
+
         if app_config.is_multi_dataset():
             # NOTE:  These routes only allow the dataset to be in the directory
             # of the dataroot, and not a subdirectory.  We may want to change
             # the route format at some point
             for dataroot_dict in server_config.multi_dataset__dataroot.values():
                 url_dataroot = dataroot_dict["base_url"]
                 bp_api = Blueprint(
-                    f"api_dataset_{url_dataroot}", __name__, url_prefix=f"/{url_dataroot}/<dataset>" + api_version
+                    f"api_dataset_{url_dataroot}", __name__,
+                    url_prefix=f"{backend_path}/{url_dataroot}/<dataset>" + api_version
                 )
                 resources = get_api_resources(bp_api, url_dataroot)
                 self.app.register_blueprint(resources.blueprint)
                 self.app.add_url_rule(
-                    f"/{url_dataroot}/<dataset>/",
+                    f"{backend_path}/{url_dataroot}/<dataset>/",
                     f"dataset_index_{url_dataroot}",
                     lambda dataset, url_dataroot=url_dataroot: dataset_index(url_dataroot, dataset),
                     methods=["GET"],
                 )
                 self.app.add_url_rule(
-                    f"/{url_dataroot}/<dataset>/static/<path:filename>",
+                    f"{backend_path}/{url_dataroot}/<dataset>/static/<path:filename>",
```

Comment:
Are you sure that the backend_path should be used here? Are the static assets served from the API URL?

Reply:
Thanks, yes this is a mistake.   Actually not really sure we even need this rule, since static assets shouldn't depend on the dataset.   We server static assets out of /static/cellgene/....
I'll revert this.

Reply:
Ah, we definitely need this rule. This is for folks who are hosting cellxgene on their own and without this, the static assets are not accessible if they use a proxy.